src/lando/main/api.py

```diff
@@ -0,0 +1,169 @@
+import datetime
+import logging
+from typing import Annotated, Literal, Union
+
+from django.contrib.auth.models import User
+from django.db import transaction
+from ninja import (
+    NinjaAPI,
+    Schema,
+)
+from ninja.responses import codes_4xx
+from ninja.security import HttpBearer
+from pydantic import Field
+
+from lando.main.models.automation_job import (
+    AutomationAction,
+    AutomationJob,
+)
+from lando.main.models.landing_job import LandingJobStatus
+from lando.main.models.repo import Repo
+
+logger = logging.getLogger(__name__)
+
+
+class APIPermissionDenied(Exception):
+    """Custom exception type to allow JSON responses for invalid auth."""
+
+    pass
+
+
+class HeadlessAPIAuthentication(HttpBearer):
+    """Authentication class to verify API token."""
+
+    def authenticate(self, request, token: str) -> str:
```

Comment:
I think it would be more ideal if we could do proper authentication with only a token (i.e., fetch a user based on a valid token, which is typically what tokens are able to do.)

This could be done by means of adding a tokens table, and you could also possibly leverage Django's `check_password` and `make_password` [functions](https://docs.djangoproject.com/en/5.1/topics/auth/passwords/#module-django.contrib.auth.hashers) for hashing.